metapool/calculate_coefficients.py

```diff
@@ -0,0 +1,147 @@
+    table_synthetic_hits_long_with_totals_beta.columns = ["plasmid_id",
+                                                          "sample_name_r",
+                                                          "count"]
+    table_synthetic_hits_long_with_totals = \
+        table_synthetic_hits_long_with_totals_beta.merge(
+            table_synthetic_hit_totals, on="sample_name_r")
+
+    # Merge updated feature-table with sample-pool information
+    table_synthetic_hits_long_with_totals_and_pools = \
+        table_synthetic_hits_long_with_totals.merge(
+            metadata_pools, on="sample_name_r")
+
+    # Aggregate across lanes (not needed)
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes = \
+        table_synthetic_hits_long_with_totals_and_pools
+
+    # Calculate counts per million
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+        "counts_per_million"] = (
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "count"] /
+            table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+                "read_count_total"]
+        ) * 1000000
+
+    # Calculate percentages
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes["percentage"] = (
+       table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+          "count"] /
+       table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+          "read_count_total"]
+    ) / 100
+
+    # Calculate log10 of counts per million
+    table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+        "counts_per_million_log10"] = np.log10(
+        table_synthetic_hits_long_with_totals_and_pools_all_lanes[
+            "counts_per_million"])
+
+    # Merge table with plasmid dilution information
+    table_synthetic_hits_with_dilutions = \
+        table_synthetic_hits_long_with_totals_and_pools_all_lanes.merge(
+            dilutions, on=["plasmid_id", "pool"])
+
+    final_table_1 = table_synthetic_hits_with_dilutions
+
+    dilution_log = final_table_1["dilution_id"].copy()
+    # TODO FIXME HACK:  Check that the dilutions we aim for are 5%, .5%, .05%
+    #  ... or this calculation is wrong!
+    dilution_log[dilution_log == 1] = 1 / 20
+    dilution_log[dilution_log == 2] = 0.1 / 20
+    dilution_log[dilution_log == 3] = 0.01 / 20
+    dilution_log[dilution_log == 4] = 0.001 / 20
+    dilution_log[dilution_log == 5] = 0.0001 / 20
+    final_table_1["dilution_log"] = np.log10(dilution_log)
+
+    if plot_fit:
+        plot_with_fit(
+            "counts_per_million_log10",
+            "dilution_log",
+            final_table_1.groupby(["pool", "sample_name"]),
+            xmax=4,
+            ymax=4
+        )
+
+    result = {
+        "pool": [],
+        "sample_name_pool": [],
+        "a_intercept": [],
+        "b_slope": []
+    }
+
+    for name, group in final_table_1.groupby(["pool", "sample_name_pool"]):
+        m, b, r, p, std_err = scipy.stats.linregress(
```

Comment:
just to confirm, `r, p, std_err` are fine to ignore, right?

Reply:
They aren't used in the current implementation, but of course if the number of samples used to determine the model is too small, or the model fit is absolutely terrible, the r and p values would indicate that.